core/src/main/java/com/netflix/conductor/core/execution/AsyncSystemTaskExecutor.java

```diff
@@ -194,11 +194,30 @@ public void execute(WorkflowSystemTask systemTask, String taskId) {
             }
             // if the current task execution has completed, then the workflow needs to be evaluated
             if (hasTaskExecutionCompleted) {
-                workflowExecutor.decide(workflowId);
+                forScheduleTaskEvaluate(task, queueName, workflowId);
```

Comment:
Hi @ghkdwlgns612 , When multiple updates are coming at the same time for different tasks. The lock acquisition will fail. However, when next time the sweeper runs it must solve the issue and schedule the simple task. If that is not happening that actual issue is that.

Reply:
Thank you for your response, let me check more:)